test/mocks/config/mocks.cc

```diff
@@ -9,6 +9,22 @@
 namespace Envoy {
 namespace Config {
 
+MockSubscriptionFactory::MockSubscriptionFactory() {
+  ON_CALL(*this, subscriptionFromConfigSource(_, _, _, _))
+      .WillByDefault(testing::Invoke([this](const envoy::api::v2::core::ConfigSource&,
+                                            absl::string_view, Stats::Scope&,
+                                            SubscriptionCallbacks& callbacks) -> SubscriptionPtr {
+        auto ret = std::make_unique<testing::NiceMock<MockSubscription>>();
+        subscription_ = ret.get();
```

Comment:
I will be surprised when I call `subscriptionFromConfigSource()` twice :)
But it's good enough as a default behavior

Reply:
Yeah, mocks typically have this simplistic default, tests that do more complicated things get to do the heavier lifting internally. Ideally mock implementations are super trivial, which this is.